turbo/stages/headerdownload/header_algos.go

```diff
@@ -643,6 +643,8 @@ func (hd *HeaderDownload) SaveExternalAnnounce(hash common.Hash) {
 }
 
 func (hd *HeaderDownload) getLink(linkHash common.Hash) (*Link, bool) {
+	hd.lock.RLock()
```

Comment:
Lock needs to go to `RecoverFromDb` instead. Here it may cause deadlocks. My convention was that un-exported functions do not lock, only exported ones (with name starting with a capital letter)